src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -87,90 +102,64 @@ public JavaGapicSurfaceTestTransformer(GapicCodePathMapper javaPathMapper) {
+    String outputPath = pathMapper.getOutputPath(service, context.getApiConfig());
+    SurfaceNamer namer = context.getNamer();
+    String name = namer.getSmokeTestClassName(service);
 
-  private SurfaceTransformerContext createContext(Interface service, ApiConfig apiConfig) {
-    ModelTypeTable typeTable =
-        new ModelTypeTable(
-            new JavaTypeTable(apiConfig.getPackageName()),
-            new JavaModelTypeNameConverter(apiConfig.getPackageName()));
-    return SurfaceTransformerContext.create(
-        service, apiConfig, typeTable, new JavaSurfaceNamer(apiConfig.getPackageName()));
+    SmokeTestClassView testClass =
+        SmokeTestClassView.newBuilder()
+            .packageName(context.getApiConfig().getPackageName())
+            .apiSettingsClassName(namer.getApiSettingsClassName(service))
+            .apiClassName(namer.getApiWrapperClassName(service))
+            .name(name)
+            .outputPath(namer.getSourceFilePath(outputPath, name))
+            .templateFileName(SMOKE_TEST_TEMPLATE_FILE)
+            .method(createSmokeTestMethodView(context))
+            // Imports must be done as the last step to catch all imports.
+            .imports(importTypeTransformer.generateImports(context.getTypeTable().getImports()))
+            .build();
+    return testClass;
   }
 
-  @Override
-  public List<String> getTemplateFileNames() {
-    List<String> fileNames = new ArrayList<>();
-    fileNames.add(TEST_TEMPLATE_FILE);
-    fileNames.add(MOCK_SERVICE_IMPL_FILE);
-    fileNames.add(MOCK_SERVICE_FILE);
-    return fileNames;
-  }
+  private TestMethodView createSmokeTestMethodView(SurfaceTransformerContext context) {
+    Method method = context.getInterfaceConfig().getSmokeTestConfig().getMethod();
+    SurfaceNamer namer = context.getNamer();
 
-  private void addTestImports(SurfaceTransformerContext context) {
-    ModelTypeTable typeTable = context.getTypeTable();
-    typeTable.saveNicknameFor("org.junit.After");
-    typeTable.saveNicknameFor("org.junit.AfterClass");
-    typeTable.saveNicknameFor("org.junit.Assert");
-    typeTable.saveNicknameFor("org.junit.Before");
-    typeTable.saveNicknameFor("org.junit.BeforeClass");
-    typeTable.saveNicknameFor("org.junit.Test");
-    typeTable.saveNicknameFor("java.io.IOException");
-    typeTable.saveNicknameFor("java.util.ArrayList");
-    typeTable.saveNicknameFor("java.util.Arrays");
-    typeTable.saveNicknameFor("java.util.List");
-    typeTable.saveNicknameFor("com.google.api.gax.testing.MockServiceHelper");
-    typeTable.saveNicknameFor("com.google.api.gax.testing.MockGrpcService");
-    typeTable.saveNicknameFor("com.google.api.gax.core.PageAccessor");
-    typeTable.saveNicknameFor("com.google.common.collect.Lists");
-    typeTable.saveNicknameFor("com.google.protobuf.GeneratedMessage");
-  }
+    ApiMethodType methodType = ApiMethodType.FlattenedMethod;
+    if (context.asMethodContext(method).getMethodConfig().isPageStreaming()) {
```

Comment:
If this function (createSmokeTestMethodView) is limited to a single method, it should probably take a method context as its parameter instead of a surface context.